vector_double.cpp

```diff
@@ -0,0 +1,207 @@
+#include "vector_double.hpp"
+
+
+template <typename T>
+Vector<T>::Vector()
+{
+    size_v = 0;
+    space  = 0;
+    elem   = nullptr;
+
+}
+
+
+// Copy Constructor
+template<class T>
+Vector<T>::Vector(const Vector<T>& b)
+{
+    space = b.space;
+    size_v = b.size_v;
+    elem = b.elem;
+
+}
+template<class T>
+Vector<T>::Vector(Vector<T>&& obj)
+{
+    space = obj.space;
+    size_v = obj.size_v;
+    elem = obj.elem;
+}
+
+//explicit
+template<class T>
+Vector<T>::Vector(unsigned int)
+{
+    size_v = 0;
+    space  = 0;
+    elem   = nullptr;
+}
+
+
+template<class T>
+Vector<T>::~Vector()
+{
+    delete [] elem;
+}
+
+template<class T>
+int Vector<T>:: size() const
+{
+    return size_v;
+}
+
+template<class T>
+int Vector<T>:: capacity()const
+{
+    return space;
+}
+
+template<typename T>
+void Vector<T>::push_back(const T &v)
+{
+
+    if (size_v >= space)
+    {
+        reserve(space +5);
+        elem [size_v++] = v;
+    }
+
+}
+
+template<class T>
+void Vector<T>::pop_back()
```

Comment:
this should return the T that was removed

Reply:
Actually David, ``pop_back()`` just removes the last element.
Straight from ``std::vector::pop_back()``: *Removes the last element in the vector, effectively reducing the container size by one.*